Further reduce use of WTF_ALLOW_UNSAFE_BUFFER_USAGE in WebCore/platform/graphics

cdumez opened the pull request:
#### 04745f32fe682525984a6ccc94dddad7a490decb
<pre>
Further reduce use of WTF_ALLOW_UNSAFE_BUFFER_USAGE in WebCore/platform/graphics
<a href="https://bugs.webkit.org/show_bug.cgi?id=285781">https://bugs.webkit.org/show_bug.cgi?id=285781</a>

Reviewed by Darin Adler.

* Source/WebCore/platform/graphics/Region.cpp:
(WebCore::Region::rects const):
(WebCore::Region::contains const):
(WebCore::Region::Shape::compareShapes):
(WebCore::Region::Shape::canCoalesce):
(WebCore::Region::Shape::appendSpan):
(WebCore::Region::Shape::appendSpans):
(WebCore::Region::Shape::segments const):
(WebCore::operator&lt;&lt;):
(WebCore::Region::Shape::bounds const):
(WebCore::Region::Shape::shapeOperation):
(WebCore::Region::Shape::spans_begin const): Deleted.
(WebCore::Region::Shape::spans_end const): Deleted.
(WebCore::Region::Shape::segments_begin const): Deleted.
(WebCore::Region::Shape::segments_end const): Deleted.
* Source/WebCore/platform/graphics/Region.h:
(WebCore::Region::Shape::spans const):
* Source/WebCore/platform/graphics/SurrogatePairAwareTextIterator.h:
(WebCore::SurrogatePairAwareTextIterator::SurrogatePairAwareTextIterator):
(WebCore::SurrogatePairAwareTextIterator::consume):
(WebCore::SurrogatePairAwareTextIterator::remainingCharacters const):
(WebCore::SurrogatePairAwareTextIterator::characters const):
* Source/WebCore/platform/graphics/UnitBezier.h:
* Source/WebCore/platform/graphics/VelocityData.cpp:
* Source/WebCore/platform/graphics/VelocityData.h:

Canonical link: <a href="https://commits.webkit.org/288756@main">https://commits.webkit.org/288756@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/99e584962b60afce8f6787b279c07b4519d756ad

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/84323 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/3947 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/38628 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/89401 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/35332 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/86408 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/4033 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/11922 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/65581 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/23415 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/87369 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/3025 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/76613 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/45874 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/2976 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/30843 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/34381 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/73870 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/31611 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/90782 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/11590 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/8418 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/74030 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/11817 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/72438 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/73228 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/17557 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/16000 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/2955 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13053 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/11542 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/17018 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/11391 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/14867 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/13164 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/platform/graphics/Region.cpp:
```diff
@@ -291,89 +286,53 @@ void Region::Shape::appendSpan(int y)
-    // Since the segments are equal the second segment can just be ignored.
-    return true;
+    auto lastSpan = m_segments.subspan(m_spans.last().segmentIndex);
+    return equalSpans(segments, lastSpan);
```
```suggestion
    return equalSpans(segments, m_segments.subspan(m_spans.last().segmentIndex));
```

cdumez committed:
Further reduce use of WTF_ALLOW_UNSAFE_BUFFER_USAGE in WebCore/platform/graphics
https://bugs.webkit.org/show_bug.cgi?id=285781

Reviewed by Darin Adler.

* Source/WebCore/platform/graphics/Region.cpp:
(WebCore::Region::rects const):
(WebCore::Region::contains const):
(WebCore::Region::Shape::compareShapes):
(WebCore::Region::Shape::canCoalesce):
(WebCore::Region::Shape::appendSpan):
(WebCore::Region::Shape::appendSpans):
(WebCore::Region::Shape::segments const):
(WebCore::operator<<):
(WebCore::Region::Shape::bounds const):
(WebCore::Region::Shape::shapeOperation):
(WebCore::Region::Shape::spans_begin const): Deleted.
(WebCore::Region::Shape::spans_end const): Deleted.
(WebCore::Region::Shape::segments_begin const): Deleted.
(WebCore::Region::Shape::segments_end const): Deleted.
* Source/WebCore/platform/graphics/Region.h:
(WebCore::Region::Shape::spans const):
* Source/WebCore/platform/graphics/SurrogatePairAwareTextIterator.h:
(WebCore::SurrogatePairAwareTextIterator::SurrogatePairAwareTextIterator):
(WebCore::SurrogatePairAwareTextIterator::consume):
(WebCore::SurrogatePairAwareTextIterator::remainingCharacters const):
(WebCore::SurrogatePairAwareTextIterator::characters const):
* Source/WebCore/platform/graphics/UnitBezier.h:
* Source/WebCore/platform/graphics/VelocityData.cpp:
* Source/WebCore/platform/graphics/VelocityData.h:

Canonical link: https://commits.webkit.org/288756@main
